src/Umbraco.Web.UI.Client/src/common/directives/components/content/edit.controller.js

```diff
@@ -197,36 +197,31 @@
 
     // This is a helper method to reduce the amount of code repitition for actions: Save, Publish, SendToPublish
     function performSave(args) {
-      var deferred = $q.defer();
-
+      
       $scope.page.buttonGroupState = "busy";
 
-      contentEditingHelper.contentEditorPerformSave({
-        statusMessage: args.statusMessage,
-        saveMethod: args.saveMethod,
-        scope: $scope,
-        content: $scope.content,
-        action: args.action
-      }).then(function (data) {
-        //success            
-        init($scope.content);
-        syncTreeNode($scope.content, data.path);
-
-        $scope.page.buttonGroupState = "success";
-
-        deferred.resolve(data);
-      }, function (err) {
-        //error
-        if (err) {
-          editorState.set($scope.content);
-        }
-
-        $scope.page.buttonGroupState = "error";
-
-        deferred.reject(err);
-      });
-
-      return deferred.promise;
+        return contentEditingHelper.contentEditorPerformSave({
+            saveMethod: args.saveMethod,
+            scope: $scope,
+            content: $scope.content,
+            action: args.action
+        }).then(function(data) {
+                //success
+                init($scope.content);
+                syncTreeNode($scope.content, data.path);
+
+                $scope.page.buttonGroupState = "success";
+                return $q.when(data);
```

Comment:
@warrenbuckley it's because of our old crappy angular version, you are correct that $q.resolve should be what we use but our version of angular doesn't have that, newer ones do but essentially it's exactly the same as $.when, they added $q.resolve because like you, people were confused about why it didn't exist.